app/models/queen.rb

```diff
@@ -1,2 +1,9 @@
 class Queen < Piece
+  def valid_move?(destination_x, destination_y)
+    return false unless destination_x.between?(0, 7) && destination_y.between?(0, 7)
+    difference_x = destination_x - position_x
+    difference_y = destination_y - position_y
+    return false if difference_x.zero? && difference_y.zero?
+    return true if difference_x.abs <= 7 && difference_y.abs <= 7
```

Comment:
On the last line of a Ruby method, `return` is not needed (it will always return the last value).

You can change this last line to `difference_x.abs <= 7 && difference_y.abs <= 7` and it will work the same.